includes/documents/class-wcpdf-invoice.php

```diff
@@ -103,25 +103,58 @@ public function init_number() {
 			return $invoice_number;
 		}
 
-		$number_store_method = WPO_WCPDF()->settings->get_sequential_number_store_method();
-		$number_store_name = apply_filters( 'wpo_wcpdf_document_sequential_number_store', 'invoice_number', $this );
-		$number_store = new Sequential_Number_Store( $number_store_name, $number_store_method );
-		// reset invoice number yearly
-		if ( isset( $this->settings['reset_number_yearly'] ) ) {
-			$current_year = date("Y");
-			$last_number_year = $number_store->get_last_date('Y');
-			// check if we need to reset
-			if ( $current_year != $last_number_year ) {
-				$number_store->set_next( apply_filters( 'wpo_wcpdf_reset_number_yearly_start', 1, $this ) );
+		$store_name     = $this->get_store_name();
+		$number_store   = $this->get_sequential_number_store( $store_name );
+		$invoice_number = $number_store->increment( $this->order_id, $this->get_date()->date_i18n( 'Y-m-d H:i:s' ) );
+
+		$this->set_number( $invoice_number );
+
+		return $invoice_number;
+	}
+
+	public function get_store_name() {
+		$document_date       = $this->get_date()->date_i18n( 'Y' );
+		$current_date        = date( 'Y' );
+		$reset_numeration    = isset( $this->settings['reset_number_yearly'] ) ? true : false;
+		$current_store_name  = "{$this->slug}_number_{$document_date}";
+
+		// reset: on
+		if( $reset_numeration ) {
+			$store_name   = $current_store_name;
+			$number_store = $this->get_sequential_number_store( $store_name );
+			$next_number  = $number_store->get_next();
+			$number_store->set_next( apply_filters( 'wpo_wcpdf_reset_number_yearly_start', $next_number, $this ) );
+
+		// reset: off
+		} else {
+			$store_name   = "{$this->slug}_number_{$current_date}";
+			$store_exists = Sequential_Number_Store::store_name_exists( $store_name );
+
+			// if store don't exist it's a new year
+			if( ! $store_exists ) {
+				$number_store = $this->get_sequential_number_store( $current_store_name );
+				$next_number  = $number_store->get_next();
+
+				// set next number based on current store max ID
+				$number_store = $this->get_sequential_number_store( $store_name );
+				$number_store->set_next( apply_filters( 'wpo_wcpdf_reset_number_yearly_start', $next_number, $this ) );
 			}
 		}
 
-		$invoice_date = $this->get_date();
-		$invoice_number = $number_store->increment( $this->order_id, $invoice_date->date_i18n( 'Y-m-d H:i:s' ) );
+		return apply_filters( "wpo_wcpdf_{$this->slug}_store_name", $store_name, $this );
+	}
 
-		$this->set_number( $invoice_number );
+	public function get_sequential_number_store( $store_name ) {
+		global $wpdb;
+		$method = isset( WPO_WCPDF()->settings->debug_settings['calculate_document_numbers'] ) ? 'calculate' : 'auto_increment';
```

Comment:
default should be `auto_increment` (this is the plugin default after all!)

Reply:
And isn't like that?

Reply:
Is using isset, I believe when it's checked is `calculate`.

Reply:
correct, I was sleeping! 🤐